src/esssans/beam_center_finder.py

```diff
@@ -34,24 +35,68 @@
 )
 
 
-def center_of_mass(data: sc.DataArray) -> sc.Variable:
+def _xy_extrema(pos: sc.Variable) -> sc.Variable:
+    x_min = pos.fields.x.min()
+    x_max = pos.fields.x.max()
+    y_min = pos.fields.y.min()
+    y_max = pos.fields.y.max()
+    return sc.concat([x_min, x_max, y_min, y_max], dim='extremes')
+
+
+def beam_center_from_center_of_mass(
+    data: MaskedData[SampleRun],
+    graph: ElasticCoordTransformGraph,
+) -> BeamCenter:
     """
-    Find the center-of-mass of the data counts.
+    Estimate the beam center via the center-of-mass of the data counts.
+
     The center-of-mass is simply the weighted mean of the positions.
+    Areas with low counts are excluded from the center of mass calculation, as they
+    typically fall into asymmetric regions of the detector panel and would thus lead
+    to a biased result. The beam is assumed to be roughly aligned with the Z axis.
+    The returned beam center is the component normal to the bema direction, projected
+    onto the X-Y plane.
 
     Parameters
     ----------
     data:
-        The data to find the center-of-mass of.
+        The data to find the beam center of.
+    graph:
```

Comment:
Should the graph be an input argument? Seems a bit strange to feed in the entire graph here, with Q and two_theta, when all we want is the `incident_beam`?
Maybe we should break up the sans graph into smaller pieces, and import it inside the function?

Reply:
> Wouldn't be against everything we wanted to avoid with Sciline?

Yes, I guess you are right